### ISSUE TITLE
useSWR hook and cache policies

### ISSUE THREAD

**Issue by a35026** [milliHQ/terraform-aws-next-js#236]
I'm using Vercel's useSWR hook for fetching page view counts  
It has its own internal cache mechanism which allows mutating it globally on the app level  
So, when I visit the page, it's triggered and mutated automatically  
```typescript
export function ViewsCounter(args: ViewsInterface) {
  const { data, mutate } = useSWR(`/api/views/${slug}`, fetcher);
  const views = new Number(data?.total);

  useEffect(() => {
    fetch(`/api/views/${slug}`, { method: "POST" })
      .then(result => result.json())
      .then((total: ViewInterface['total']) => mutate(total))
      .catch(console.log);
  }, [slug]);

  if (!show) return null;
  return (
    <>
      {views > 0
        ? new Intl.NumberFormat('en-US', {
            notation: "compact",
            compactDisplay: "short"
          }).format(data?.total)
        : 0
      }
    </>
  );
}
```

All this beauty works great in production with cache disabled or default AWS managed policy "Amplify"  
But doesn't work with the policy `tf-website_tfn-cache` added and managed by `Terraform Next.js`  

The Amplify policy has minimum, maximum and default TTLs of 2, 600 and 2 respectively  
The Terraform Next.js policy has 0, 31536000 and 86400

Not sure why the latter policy doesn't allow SWR cache mechanism to work as expected in the first place  
So, the question is how can I control TTLs for the policy imposed by Terraform Next.js?  

I am using default Terraform Next.js main.tf template for custom domains (which works flawlessly)

**Comment by ofhouse** [comment]
Hi,
yeah the default TTL is too high and should probably `0` (Not cached) instead of `86400` (24h).

For explanation what the TTL values here mean:
- **minimum**: 0s (Amplify uses 2s)
  The minimum time a response should live in the CloudFront cache before it gets invalidated
- **maximum**: 31536000s (Amplify uses 600s)
  The maximum time a response should live in the CloudFront cache before it gets invalidated
- **default**: 86400s (Amplify uses 2s)
  When a response doesn't include a `Cache-Control` header, CloudFront uses the value from here to control how long the response should live in the CloudFront cache before it gets invalidated

You can however workaround this by setting the Cache-Control header to `0` in the API response, like this:
```diff
// pages/api/views/[...slug].js

export default function handler(req, res) {
  ...
+ res.setHeader('Cache-Control', 'max-age=0');
  res.end()
}
```

This way CloudFront would respect the Cache-Control header and the response would not be cached (`max-age=0`) at all.
It would also be a good case for using [middleware](https://nextjs.org/docs/middleware) but unfortunately we currently don't support using middleware.

**Comment by ofhouse** [comment]
This has been fixed in [`v0.11.0`](https://github.com/milliHQ/terraform-aws-next-js/releases/tag/v0.11.0).
